src/meta/GflagsManager.cpp

```diff
@@ -23,19 +25,60 @@ std::string GflagsManager::gflagsValueToThriftValue<std::string>(
     return flag.current_value;
 }
 
-void GflagsManager::declareGflags() {
-    // declare all gflags in ClientBasedGflagsManager
+std::unordered_map<std::string, cpp2::ConfigMode> GflagsManager::parseConfigJson() {
+    auto path = fs::FileUtils::readLink("/proc/self/exe").value();
+    auto json = fs::FileUtils::dirname(path.c_str()) + "/../share/resources/gflags.json";
+    std::unordered_map<std::string, cpp2::ConfigMode> configModeMap;
+    Configuration conf;
+    if (!conf.parseFromFile(json).ok()) {
+        LOG(ERROR) << "Load gflags json failed";
+        return configModeMap;
+    }
+    std::vector<std::string> keys = {"IMMUTABLE", "REBOOT", "MUTABLE", "IGNORED"};
+    std::vector<cpp2::ConfigMode> modes = {cpp2::ConfigMode::IMMUTABLE, cpp2::ConfigMode::REBOOT,
+                                           cpp2::ConfigMode::MUTABLE, cpp2::ConfigMode::IGNORED};
+    for (size_t i = 0; i < keys.size(); i++) {
+        std::vector<std::string> values;
+        if (!conf.fetchAsStringArray(keys[i].c_str(), values).ok()) {
+            continue;
+        }
+        cpp2::ConfigMode mode = modes[i];
+        for (const auto& name : values) {
+            configModeMap[name] = mode;
+        }
+    }
+    return configModeMap;
+}
+
+std::vector<cpp2::ConfigItem> GflagsManager::declareGflags(const cpp2::ConfigModule& module) {
+    std::vector<cpp2::ConfigItem> configItems;
+    if (module == cpp2::ConfigModule::UNKNOWN) {
+        return configItems;
+    }
+    auto configModeMap = parseConfigJson();
     std::vector<gflags::CommandLineFlagInfo> flags;
     gflags::GetAllFlags(&flags);
     for (auto& flag : flags) {
         auto& name = flag.name;
         auto& type = flag.type;
         cpp2::ConfigType cType;
-        // default config type will take effect after reboot
-        cpp2::ConfigMode mode = cpp2::ConfigMode::REBOOT;
         VariantType value;
         std::string valueStr;
 
+        // default config type would be immutable
+        cpp2::ConfigMode mode = cpp2::ConfigMode::IMMUTABLE;
+        if (configModeMap.find(name) != configModeMap.end()) {
+            mode = configModeMap[name];
+        }
+        // ignore some useless gflags
+        if (mode == cpp2::ConfigMode::IGNORED) {
```

Comment:
mode default is cpp2::ConfigMode::IMMUTABLE, if configModeMap.find(name) != configModeMap.end() is false, this condition never true, is it right to go the next logic?

Reply:
The configModeMap is load from json, if found in json, replace the default mode. So for those configs we want to ignore, the mode will set to  cpp2::ConfigMode::IGNORED, and we can just ignore those gflags.